src/test/searchqueryparsertest.cpp

```diff
@@ -3,14 +3,16 @@
 #include <QDir>
 #include <QTemporaryFile>
 
+#include "test/librarytest.h"
+
 #include "library/searchqueryparser.h"
 #include "util/assert.h"
 
-class SearchQueryParserTest : public testing::Test {
+class SearchQueryParserTest : public LibraryTest {
   protected:
     SearchQueryParserTest()
             : m_database(QSqlDatabase::addDatabase("QSQLITE")),
-              m_parser(m_database) {
+              m_parser(collection()) {
         QTemporaryFile databaseFile("mixxxdb.sqlite");
```

Comment:
Please remove the manually created connection to a temporary database. It should be obsolete now since LibraryTest provides a TrackCollection with a database.